src/plugins/layout/stateprovider.ts

```diff
@@ -89,12 +89,20 @@ export function createElementLayoutState(element: Element): IElementLayoutState
 
   elementState.layout = null;
   if (rect) {
-    elementState.layout = {
-      x: Math.round(rect.left),
-      y: Math.round(rect.top),
-      width: Math.round(rect.width),
-      height: Math.round(rect.height)
-    };
+    if (styles["visibility"] !== "hidden") {
+      elementState.layout = {
+        x: Math.round(rect.left),
+        y: Math.round(rect.top),
+        width: Math.round(rect.width),
+        height: Math.round(rect.height),
+        color: styles["color"]
+       };
```

Comment:
Could you not jsut do:

node.style.color and similarly for other cases to compute only the CSS properties you need and not everything?

Reply:
node.style.color only works if the color property is assigned via inline style. If a style sheet was used, it gives inaccurate results. To factor in cascading nature of style sheets, we will need computed styles.